tests/test_timestepping.py

```diff
@@ -27,8 +27,7 @@ def set(self, dt, value):
     def get(self, dt):
         return self.memory.get((self.t, dt))
 
-    def __call__(self, t):
-        dt = t - self.t
+    def __call__(self, t, dt):
```

Comment:
As `dt` is now given, we can get rid of a whole bunch of stuff in this class. `self.t` is pointless now, as well as `get` and `set`.